src/reader/mod.rs

```diff
@@ -204,7 +201,7 @@ impl InnerIndexReader {
         Ok(segment_readers)
     }
 
-    fn create_new_searcher_generation(
+    fn track_segment_readers_in_inventory(
```

Comment:
The previous name seems better to me, since we are creating a `SearcherGeneration` and making sure to do it in a way that we track it int the inventory, and the inventory tracks `SearcherGeneration` not segment readers.

Reply:
my issue with the previous name was that it's not clear that inventory is used to track and not the source of the generation